### PR TITLE
[V2] fix: Avoid calling detach for ControllerUnpublish Volume

### PR DESCRIPTION
**Pull request by nearora-msft**
**What type of PR is this?**
/kind bug

**What this PR does / why we need it**:
Issue: When a ControllerPublishVolume call comes for a replica attachment, it is moved to demoted state. If another ControllerPublishVolume  comes for the same attachment, the current implementation deletes the replica attachment because of this condition that whether the volume attachment is a replica or not becomes true here the 2nd time around: https://github.com/kubernetes-sigs/azuredisk-csi-driver/blob/467066e583ee720abc58b66a95d3332269753112/pkg/provisioner/crdprovisioner.go#L631

Fix: Remove this condition from the UnpublishVolume function and create a separate function for deatching the volume

**Which issue(s) this PR fixes**:

Fixes #

**Requirements**:
- [ ] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Special notes for your reviewer**:

**Release note**:
```
none
```

### CODE REVIEW

**Comment by edreed** [comment]
@sunpa93 Could you please take a look at this change?

File: pkg/provisioner/crdprovisioner.go [hunk 1]
```diff
@@ -627,61 +628,74 @@ func (c *CrdProvisioner) UnpublishVolume(
...
+	if azVolumeInstance.Spec.MaxMountReplicaCount == 0 {
 		// if volume's maxMountReplicaCount == 0, detach and wait for detachment to complete
-		// if AzVolumeAttachment instance indicates that previous attachment request was successful, annotate the CRI with detach request so that the underlying volume attachment can be properly detached.
-		if azVolumeAttachmentInstance.Status.Detail != nil {
-			// if detachment is pending, return to prevent duplicate request
-			if azVolumeAttachmentInstance.Status.State == diskv1beta1.Detaching {
-				return status.Errorf(codes.Aborted, "detachment still in process for volume (%s) and node (%s)", volumeName, nodeID)
-			}
+		return c.detachVolume(ctx, azVAClient, azVolumeAttachmentInstance)
+	}
```
Comment (sunpa93) [inline pkg/provisioner/crdprovisioner.go:636]: I think it would be more consistent to refactor this into a separate `demote` function.

**Pull request merged**